app/src/main/java/org/simple/clinic/enterotp/EnterOtpUiRenderer.kt

```diff
@@ -41,12 +41,20 @@ class EnterOtpUiRenderer(
     when (model.protectedState) {
       is Allowed -> {
         ui.showOtpEntryMode(OtpEntry)
-        ui.hideError()
+        generateUiForAllowingOtpEntry(model.protectedState.attemptsMade, model.protectedState.attemptsRemaining)
       }
       is Blocked -> {
         ui.showOtpEntryMode(BruteForceOtpEntryLocked(model.protectedState.blockedTill))
         ui.showLimitReachedError(model.protectedState.attemptsMade)
       }
     }
   }
+
+  private fun generateUiForAllowingOtpEntry(attemptsMade: Int, attemptsRemaining: Int) {
+    if (attemptsMade == 0) {
```

Comment:
Hmm, I think we are using magic numbers in this commit. For example, deciding which error format to show or whether to hide/show an error. Instead, can we move them to a model, so that they don't lose context?

Like, `model.hasNoIncorrectPinEntries` or `mode.hasReachedMinPinRetries` (for showing error with attempts remaining, etc.,)